stlab/forest_algorithms.hpp

```diff
@@ -56,14 +58,14 @@ struct forest_insert_iterator {
 };
 
 template <class Forest>
-auto forest_inserter(Forest& f) {
-    return forest_insert_iterator(f);
+auto make_inserter(Forest& f) {
```

Comment:
Not to ping-pong this. But the idea was to follow the standard pattern
https://en.cppreference.com/w/cpp/iterator/inserter

I'm actually not sure why std::inserter is insufficient.

Reply:
Are you asking if we should call this routine `inserter`, or we should be using `std::inserter` instead of this one? I'm okay with renaming it to `inserter`. As for using `std::inserter`, `std::insert_iterator` doesn't meet the requirements needed by my fledgling `ForestOutputIterator` concept: the latter needs to form the shape of the forest as values are being inserted, not just insert values.

Reply:
An insert iterator wouldn't move on ++. It is an output iterator.  I don't know quite what this is. It also isn't even a valid iterator from what I can see - there is no iterator_category, for example.

Reply:
I'm pretty sure I can rewrite the algorithms without this class.

Reply:
I rewrote it, and things are cleaner. Unfortunately I still need the different `insert_iterator`, because I have to be able to pivot the output iterator to maintain the forest's shape when outputting.

Note that the whole class of standard algorithms will Just Work for forests. The two-container algorithms will also Just Work, assuming the two forests have the same number of nodes (regardless of shape.) This algorithm specifically inserts nodes into a forest and builds out that forest into the same shape as its source forest - a unique issue the standard algorithms haven't had to deal with before.